In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv('../../Data/02ParaLimpiar/02desastres_definitivo.csv', encoding='utf-8', delimiter=';', engine='python')

In [3]:
drought_filter = (df['Disaster Type'] == 'Drought')
df_drought = df[drought_filter]
df_drought.head()

Dis No  Year   Seq Disaster Subgroup Disaster Type  \
0   1900-9002-CPV  1900  9002    Climatological       Drought   
1   1900-9001-IND  1900  9001    Climatological       Drought   
24  1910-9006-BFA  1910  9006    Climatological       Drought   
27  1910-9006-CPV  1910  9006    Climatological       Drought   
30  1910-9006-GMB  1910  9006    Climatological       Drought   

   Disaster Subtype       Country  ISO          Region Continent  ...  \
0           Drought    Cabo Verde  CPV  Western Africa    Africa  ...   
1           Drought         India  IND   Southern Asia      Asia  ...   
24          Drought  Burkina Faso  BFA  Western Africa    Africa  ...   
27          Drought    Cabo Verde  CPV  Western Africa    Africa  ...   
30          Drought  Gambia (the)  GMB  Western Africa    Africa  ...   

   Longitude Start Year Start Month  Start Day End Year End Month End Day  \
0        NaN       1900         NaN        NaN     1900       NaN     NaN   
1        NaN       1900         NaN        NaN     1900       NaN     NaN   
24       NaN       1910         NaN        NaN     1914       NaN     NaN   
27       NaN       1910         NaN        NaN     1914       NaN     NaN   
30       NaN       1910         NaN        NaN     1914       NaN     NaN   

    Total Deaths  Total Affected  Total Damages Adj  
0        11000.0             NaN                NaN  
1      1250000.0             NaN                NaN  
24           NaN             NaN                NaN  
27           NaN             NaN                NaN  
30           NaN             NaN                NaN  

[5 rows x 26 columns]

In [5]:
df_drought.shape

(803, 26)

In [23]:
df_drought[['Start Year', 'End Year']].isnull().sum()

Start Year    0
End Year      0
dtype: int64

In [22]:
df_drought['Duration'] = df_drought['End Year']-df_drought['Start Year']
df_drought['Duration'].value_counts()

C:\Users\blanc\AppData\Local\Temp\ipykernel_13400\2714018211.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_drought['Duration'] = df_drought['End Year']-df_drought['Start Year']


0    497
1    181
2     54
4     31
3     24
5     11
6      3
9      2
Name: Duration, dtype: int64

De los 803 eventos de sequía, más de la mitad no han llegado a un año de duración y casi una cuarta parte han durado un año

In [24]:
df_drought.shape

(803, 27)

## PIVOT TABLE: Location, Country, Region, Continent, Year, Associated Dis, Total Deaths, Total Affected, Total Damages Adj

In [32]:
pivot_table_1 = pd.pivot_table(df_drought, index= 'Year', aggfunc='size')
pivot_table_1.head()


Year
1900    2
1903    1
1906    1
1910    9
1919    1
dtype: int64

In [29]:
# Filtrar el dataset original
filtered_data = df[['Country', 'Region' ,'Continent','Associated Dis', 'Year', 'Dis Mag Value']]

# Crear pivot table considerada pivot principal
pivot_table_drought = df_drought.pivot_table(index=['Country', 'Region', 'Continent', 'Associated Dis', 'Year'], values='Duration', aggfunc='count')
pivot_table_drought.head(100)

Duration
Country      Region          Continent Associated Dis Year          
Afghanistan  Southern Asia   Asia      Food shortage  1969         1
                                                      2008         1
                                                      2011         1
                                                      2018         1
                                                      2021         1
...                                                              ...
Gambia (the) Western Africa  Africa    Food shortage  2002         1
                                                      2015         1
Ghana        Western Africa  Africa    Food shortage  1977         1
                                       Wildfire       1980         1
Greece       Southern Europe Europe    Heat wave      1990         1

[100 rows x 1 columns]

In [20]:
pivot_table_drought1 = df_drought.pivot_table(index=['Year', 'Continent', 'Region', 'Country'], columns= 'Associated Dis', values='Total Affected')
pivot_table_drought1.head(45)

Associated Dis                                                    Cold wave  \
Year Continent Region             Country                                     
1900 Africa    Western Africa     Cabo Verde                            NaN   
1920 Africa    Western Africa     Cabo Verde                            NaN   
1921 Europe    Russian Federation Soviet Union                          NaN   
1940 Africa    Western Africa     Cabo Verde                            NaN   
1942 Asia      Southern Asia      India                                 NaN   
1943 Asia      Southern Asia      Bangladesh                            NaN   
1946 Africa    Western Africa     Cabo Verde                            NaN   
1965 Asia      Southern Asia      India                                 NaN   
1966 Asia      South-Eastern Asia Indonesia                             NaN   
1973 Africa    Eastern Africa     Ethiopia                              NaN   
                                  Somalia                               NaN   
1978 Asia      South-Eastern Asia Indonesia                             NaN   
1980 Africa    Middle Africa      Chad                                  NaN   
1981 Africa    Southern Africa    Eswatini                              NaN   
1982 Asia      South-Eastern Asia Indonesia                             NaN   
1988 Africa    Eastern Africa     Madagascar                            NaN   
     Asia      Eastern Asia       China                                 NaN   
1989 Africa    Eastern Africa     Rwanda                                NaN   
1997 Asia      South-Eastern Asia Indonesia                             NaN   
     Oceania   Melanesia          Papua New Guinea                      NaN   
1998 Africa    Western Africa     Guinea                                NaN   
1999 Africa    Eastern Africa     Somalia                               NaN   
                                  Kenya                                 NaN   
                                  Uganda                                NaN   
     Americas  South America      Paraguay                             12.0   
     Asia      Southern Asia      Pakistan                              NaN   
2000 Asia      Southern Asia      India                                 NaN   
                                  Afghanistan                           NaN   
2001 Africa    Middle Africa      Angola                                NaN   
     Americas  Central America    Guatemala                             NaN   
2002 Africa    Eastern Africa     Uganda                                NaN   
                                  Malawi                                NaN   
2004 Africa    Eastern Africa     Kenya                                 NaN   
2005 Africa    Eastern Africa     Somalia                               NaN   
                                  Kenya                                 NaN   
                                  Burundi                               NaN   
2006 Asia      Eastern Asia       China                                 NaN   
2010 Africa    Eastern Africa     Somalia                               NaN   
2015 Asia      South-Eastern Asia Indonesia                            11.0   
     Oceania   Melanesia          Papua New Guinea                     24.0   
2019 Asia      Southern Asia      Pakistan                              NaN   
2020 Americas  Northern America   United States of America (the)        NaN   
2022 Africa    Eastern Africa     Uganda                                NaN   
     Americas  Northern America   United States of America (the)        NaN   

Associated Dis                                                    Crop failure  \
Year Continent Region             Country                                        
1900 Africa    Western Africa     Cabo Verde                               NaN   
1920 Africa    Western Africa     Cabo Verde                               NaN   
1921 Europe    Russian Federation Sov